### ISSUE TITLE
RectifiedAdam()

### ISSUE THREAD

**Issue by markub3327** [keras-team/keras#16195]
Hello,

 Would be interested to port `RectifiedAdam` from **tensorflow/addon** with addding exclude_from_weight_decay extension?

For more details https://github.com/tensorflow/addons/issues/2681

Thanks a lot.
Have a nice day.

**Comment by chunduriv** [comment]
@markub3327,

Can you please elaborate about your feature and please specify the use cases for this feature. Thanks!

**Comment by markub3327** [comment]
I need to exclude some parameters from the weight decay process. Useful during training the Transformer model. List of regex patterns of variables excluded from weight decay. Variables whose names contain a substring matching the pattern will be excluded.

[Here](https://github.com/microsoft/unilm/blob/master/beit/semantic_segmentation/backbone/beit.py#L415) is the use case.

The code behind exclude_from_weight_decay from tensorflow/addons is [here](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/weight_decay_optimizers.py#L245).

**Comment by haifeng-jin** [comment]
@chenmoneygithub Any comments on this?

**Comment by haifeng-jin** [comment]
The recommended way of using this is to use it in tf addons for now,
unless we see a substantial usage of it in the keras examples.

**Comment by markub3327** [comment]
@haifeng-jin 

I can create a PR on this issue. Thanks!
